src/models/db/case.js

```diff
@@ -1,72 +1,163 @@
 'use strict'
 
-module.exports = CaseFactory
-
-const Container = require('constitute').Container
-const Model = require('five-bells-shared').Model
-const UriManager = require('../../lib/uri')
-const config = require('../../lib/config')
-const PersistentKnexModelMixin = require('five-bells-shared').PersistentKnexModelMixin
-const Validator = require('five-bells-shared/lib/validator')
 const knex = require('../../lib/knex').knex
-const moment = require('moment')
-
-CaseFactory.constitute = [UriManager, Validator, Container]
-function CaseFactory (uri, validator, container) {
-  class Case extends Model {
-    static convertFromExternal (data) {
-      // ID is optional on the incoming side
-      if (data.id) {
-        data.id = uri.parse(data.id, 'case').id.toLowerCase()
-      }
+const _ = require('lodash')
+const co = require('co')
+const assert = require('assert')
 
-      data.expires_at = new Date(data.expires_at)
+const TABLE_NAME = 'N_CASES'
+const STATUS_TABLE_NAME = 'N_LU_CASE_STATUS'
 
-      return data
-    }
+function getTransaction (options) {
+  return !options ? knex : (!options.transaction ? knex : options.transaction)
+}
 
-    static convertToExternal (data) {
-      data.id = uri.make('case', data.id.toLowerCase())
-      data.notaries = [config.getIn(['server', 'base_uri'])]
-      data.expires_at = moment(data.expires_at).toISOString() // format('YYYY-MM-DD HH:mm:ss');
-      delete data.Notaries
-      if (!data.exec_cond_fulfillment) {
-        delete data.exec_cond_fulfillment
-      }
-      return data
-    }
+function transaction (callback) {
+  return knex.transaction(co.wrap(callback))
+}
 
-    static convertFromPersistent (data) {
-      data.execution_condition = JSON.parse(data.execution_condition)
-      if (data.exec_cond_fulfillment) {
-        data.exec_cond_fulfillment = JSON.parse(data.exec_cond_fulfillment)
-      }
-      data.notification_targets = JSON.parse(data.notification_targets)
-      delete data.created_at
-      delete data.updated_at
-      if (!Array.isArray(data.notaries)) {
-        data.notaries = [ data.notaries ]
+function * insertCases (caseObjects, options) {
+  if (caseObjects.length <= 0) {
+    return
+  }
+  return Promise.all(_.map(caseObjects, (caseObject) => {
+    return convertToPersistent(caseObject, options)
```

Comment:
This doesn't have to be inside the promise chain right? Should be simpler to convert first and then use promises just for the inserts.